src/providers/extraction_providers/visionExtractionProvider.ts

```diff
@@ -0,0 +1,74 @@
+import { ExtractionProvider } from './extractionProvider';
+import { Image, ExtractedImage } from '../../models/data_models/image';
+import { AssignedLabel, DetectedObject, Coordinate } from '../../models/data_models/extraction';
+
+import * as vision from '@google-cloud/vision';
+
+const enum requestTypes {
+	labelDetection = 'LABEL_DETECTION',
+	objectLocalization = 'OBJECT_LOCALIZATION',
+}
+
+export class VisionExtractionProvider implements ExtractionProvider {
+	protected createAssignedLabel(
+		label: vision.protos.google.cloud.vision.v1.IEntityAnnotation
+	): AssignedLabel {
+		return {
+			description: label?.description || '',
+			confidence: label?.score || 0,
+		};
+	}
+
+	protected createBoundingPoly(
+		boundingPoly: vision.protos.google.cloud.vision.v1.INormalizedVertex
+	): Coordinate {
+		return {
+			x: boundingPoly?.x || 0,
+			y: boundingPoly?.y || 0,
+		};
+	}
+
+	protected createDetectedObject(
+		object: vision.protos.google.cloud.vision.v1.ILocalizedObjectAnnotation
```

Comment:
oh dear

let's create a typealias

```ts
type LocalizedObjectAnnotation = vision.protos.google.cloud.vision.v1.ILocalizedObjectAnnotation
```

then we can write
```suggestion
		object: LocalizedObjectAnnotation
```

Reply:
likewise for the vertex function